test/test-pre-push.sh

```diff
@@ -529,11 +529,6 @@ begin_test "pre-push with unowned lock"
     ok="${PIPESTATUS[0]}"
     set -e
 
-    if [ "0" -eq $ok ]; then
```

Comment:
How about removing lines 527, 529, and 530 too? Based on the behavior outlined in https://github.com/git-lfs/git-lfs/issues/1884 for `lfs.<url>.locks` null values, this test should always require a successful push.